main.cpp

```diff
@@ -29,36 +29,35 @@ Create a branch named Part2
 #include <string>
 struct T
 {
-    T(int v, const char* n);   //1
+    T(int v, const char* n);   
     int value;
     std::string name;
 };
 
 T::T(int v, const char* n) : value(v), name(n) {}
 
-struct Compare                                //4
+struct Compare                                
 {
-    T* compare(T* a, T* b) //5
+    const T* compare(const T& a, const T& b) 
     {
-        if(a != nullptr && b != nullptr)
-        {
-            if( a->value < b->value ) return a;
-            if( a->value > b->value ) return b;
-        }
-        return nullptr;
+        const T* ptr {nullptr};
+        
+        if( a.value < b.value ) return (ptr = &a);
+        if( a.value > b.value ) return (ptr = &b);
+
+        return ptr;
```

Comment:
Temporary local variables are not needed here.
and never write `return a = b;`
Expressions like that are impossible to tell what value is actually being returned.
- [ ] revise